modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPhpCodegen.java

```diff
@@ -115,8 +115,8 @@ public AbstractPhpCodegen() {
         typeMapping.put("string", "string");
         typeMapping.put("byte", "int");
         typeMapping.put("boolean", "bool");
-        typeMapping.put("date", "\\DateTime");
-        typeMapping.put("Date", "\\DateTime");
+        typeMapping.put("date", "date");
```

Comment:
That is a significant change as soon as it overrides all PHP codegens(php client, lumen, laravel, symfony, slim4, php-ze). I guess it can break many unit tests.

I don't know perfectly how php client works, so I should ask @ackintosh or @wing328 to review.

Reply:
I would say that mapping format `date` and `date-time` to `\DateTime` PHP instance was a mistake at the first place, but I don't know how to solve it without breaking changes.

Problems with `\DateTime` auto converting that I can recall is:
- When client sends `date-time` value without timezone like `2010-01-01T00:00:00` it can be parsed with default server timezone and if it's different we can end up with different hours or even date.
- You don't know which date time format client used because generated code already converted it to `\DateTime` without ability to get source string.
- You always not sure that `DateTime` is equals to original request parameter because of different datetime formats and timezone(especially when TZ is omitted).

Reply:
The mapping of date-time to DateTime is OK, this is expected because it'll default to the expected format of DateTime::ATOM.

But for 'date', it needs to be 'date', otherwise the system will think that it is a DateTime and return the date and time.

This all started because the date it was returning was incorrect, it was the wrong format, it was returning a date time rather than the expected date.

The date format issue that you're describing with the timezone really seems to be more of an issue with the protocol than the generator.

The specification is very clear that the date-time format should be RFC3339 format, and the timezone issue is negated if you always use UTC.

Reply:
Imagine Public API of booking service. User wants to create appointment at 9 o'clock Berlin time. Application configured to send remind notifications 2 hours before. You suggest to always convert date time to UTC value, yes, it's doable, but it will look confusing in the database and you cannot force all companies to use autogenerated client library, so there is no guarantee that developer sends date time value in RFC3339 format.

Anyway, it's just my personal view on date time issues. I can use `type: string` without `format: datetime` if I don't like autogenerated implementations.

Reply:
I think that’s an issue with the protocol itself.

Anyway, the problem here is not with datetimes, it’s with date.

A date should be a date only, not a date time.

Reply:
Technically there is no `date` class in php, only `\DateTime`. I would map it to `string` scalar type to avoid syntax errors.

Reply:
You're right, I wasn't sure if it was a "php" type or some kind of virtual type that openapi would make up. For example, there's no such data type as "map".

I'll change it to string, as I think that makes the most sense here.